pkg/plugins/runner/sorted_test.go

```diff
@@ -0,0 +1,62 @@
+package runner
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/nornir-automation/gornir/pkg/gornir"
+	"github.com/nornir-automation/gornir/pkg/plugins/logger"
+)
+
+// TestSorted runs test func and verifies the hosts are executed
+// in alphabetical order by checking the results come in the right
+// order
+func TestSorted(t *testing.T) {
+	testCases := []struct {
+		name          string
+		expected      []string
+		sleepDuration time.Duration
+	}{
+		{
+			name:          "simple test",
+			expected:      []string{"dev1", "dev2", "dev3", "dev4"},
+			sleepDuration: 1 * time.Millisecond,
+		},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		results := make(chan *gornir.JobResult, len(testHosts))
+		t.Run(tc.name, func(t *testing.T) {
+			rnr := Sorted()
+			if err := rnr.Run(
+				context.Background(),
+				&testTaskSleep{sleepDuration: tc.sleepDuration},
+				testHosts,
+				gornir.NewJobParameters("test", logger.NewLogrus(false)),
+				results,
+			); err != nil {
+				t.Fatal(err)
+			}
+			if err := rnr.Wait(); err != nil {
+				t.Fatal(err)
+			}
+			close(results)
+
+			// let's process the results and turn it into a list so we can
+			// compare with our expected value
+			got := make([]string, len(testHosts))
+			i := 0
+			for res := range results {
+				got[i] = res.JobParameters().Host().Hostname
+				i++
+			}
+			if !cmp.Equal(got, tc.expected) {
+				t.Error(cmp.Diff(got, tc.expected))
+			}
```

Comment:
We cannot ensure the results will be returned in the same order the goroutines were launched. It might work in this case, but it won't in a more general context.

Reply:
there are no goroutines in the Sorted runner. Things are run synchronously.

Reply:
Got it, didn't notice this.